snap/quota/resources.go

```diff
@@ -55,6 +55,24 @@ type ResourceThreads struct {
 	Limit int `json:"limit"`
 }
 
+type ResourceJournalSize struct {
+	Limit quantity.Size `json:"limit"`
+}
+
+type ResourceJournalRate struct {
+	Count  int `json:"count"`
+	Period int `json:"period"`
+}
+
+// ResourceJournal represents the available journal quotas. It's structured
+// a bit different compared to the other resources to support namespace only
+// cases, where the existence of != nil ResourceJournal with empty values
+// indicates that the namespace only is wanted.
+type ResourceJournal struct {
+	Size *ResourceJournalSize `json:"size"`
```

Comment:
Do we need limit to be nested here if ResourceJournalSize is just a single value? Is it because we want it to be extendable in the future in case of additional size constraints?

Reply:
Yea it's only because the Size is an optional limit, I can however see I forgot to add 'omitempty'